lib/matplotlib/axes/_base.py

```diff
@@ -513,11 +513,19 @@ def _plot_args(self, axes, tup, kwargs, *,
 
         label = kwargs.get('label')
         n_datasets = max(ncx, ncy)
-        if n_datasets > 1 and not cbook.is_scalar_or_string(label):
+        if not cbook.is_scalar_or_string(label):
```

Comment:
For better readability I would reverse the logic:

```
if cbook.is_scalar_or_string(label):
    labels = [label] * n_datasets
else:
    ...
```

Reply:
Ah yes.  I didn't see the forest for the trees.  Done.